packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -657,6 +662,15 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
 
   @override
   Widget build(BuildContext context) {
+    switch (Theme.of(context).platform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+        _vibrate = true;
+        break;
+      case TargetPlatform.iOS:
+        _vibrate = false;
+        break;
+    }
```

Comment:
For my own education, what's the difference? This only changes when set state is called I assume? If it were possible to keep the modal dialog on while changing the platform in the gallery, would it indirectly trigger a set state or no?